File: mobile/lib/features/recette/domain/entities/encaissement.dart
enum ModeEncaissement {
  especes,
  mobileMoney;

  static ModeEncaissement fromJson(String value) => switch (value) {
        'ESPECES' => especes,
        'MOBILE_MONEY' => mobileMoney,
        _ => especes,
      };

  String toJson() => switch (this) {
        especes => 'ESPECES',
        mobileMoney => 'MOBILE_MONEY',
      };

  String get label => switch (this) {
        especes => 'Espèces',
        mobileMoney => 'Mobile Money',
      };
}

class Encaissement {
  final int? id;
  final int ligneRecetteId;
  final int? operationFinanciereId;
  final double montant;
  final ModeEncaissement modeEncaissement;
  final DateTime dateEncaissement;
  final String? reference;
  final String? commentaire;

  const Encaissement({
    this.id,
    required this.ligneRecetteId,
    this.operationFinanciereId,
    required this.montant,
    required this.modeEncaissement,
    required this.dateEncaissement,
    this.reference,
    this.commentaire,
  });
}

File: backend/src/main/java/com/tmk/vtcmanager/application/domain/recette/Encaissement.java
package com.tmk.vtcmanager.application.domain.recette;

import com.tmk.vtcmanager.application.domain.operation.ModePaiement;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Encaissement {

    private Long id;
    private Long ligneRecetteId;
    private Long operationFinanciereId;
    private BigDecimal montant;
    private ModePaiement modeEncaissement;
    private LocalDate dateEncaissement;
    private String reference;
    private String commentaire;

    /** Annulé (jamais supprimé) : ignoré des recalculs et des agrégats. */
    private LocalDateTime annuleLe;
    private String annulePar;
    private String motifAnnulation;
}
